main.rb

```diff
@@ -6,57 +6,60 @@ def main
     puts welcome_screen
 
     print "Please enter the dimension of the table [default: 3 (3 means 3x3)]: "
-    dimension = gets.strip.to_i
 
-    player_name1,player1_symbol = setup_player(1)
+    dimension = gets.strip.to_i
 
-    player_name2,player2_symbol = setup_player(2)
+    game = Game.new(dimension)
 
-    game = Game.new(dimension, player_name1, player1_symbol, player_name2, player2_symbol)
+    player_name1,player1_symbol = get_player_info(1)
+    game.set_player(0, player_name1, player1_symbol)
 
-    on_game = true
-    play_again = true
+    player_name2,player2_symbol = get_player_info(2)
+    game.set_player(1, player_name2, player2_symbol)
 
     print_table(game)
+    
+    while game.still_active?(game.status)
 
-    while on_game
+        save_new_move(game)
 
-      game_status = game.status
+        system("cls") || system("clear")
 
-      if game.still_active?(game_status)
-        next_move = setup_move(game.player_on_turn, game.board)
-        game.make_a_move(next_move)
-        game.fill_table
+        print_table(game)
 
-        system("cls") || system("clear")
         game.switch_players
 
-        print_table(game)
-      else
-        if game.resolve_game?(game_status)
-          puts "#{game.player_on_turn.name} you won!!"
-        else
-          puts "Sorry none of you won"
-        end
-          on_game = false
-      end
     end
 
-    print 'Do you want to play again? (y/n) [default: n]: '
-    play_again = gets.chomp.downcase.match(/y|yes/) ? true : false
-    on_game = true if play_again
-    system("cls") || system("clear") if play_again
-    break unless play_again
+    announce_winner(game)
+
+    break unless play_again?
+
+  end
+  # goodbye
+  puts "Thanks for playing this game!"
+end
+
+def save_new_move game
+  next_move = setup_move(game.player_on_turn, game.board)
+  game.make_a_move(next_move)
+  game.fill_table
+end
```

Comment:
@heyapricot what would be `current_player`? a variable with `game.player_on_turn` assigned on it or a function (in the same `main` file) that calls `game.player_on_turn`? I think the object's dependencies you mention are not related to the Facade Pattern, it is related about to make a clear separation between the core application and the interface we are using as a source of input (in our case the `main` file). If we use another facade (for the web, for example) the code will be almost the same we have for the `main` file just with a different source of input (text fields and some things like that) but things like `game.player_on_turn` won't break the pattern 
because that doesn't affect the separation of input sources.

Reply:
👍